fedimint-core/src/lib.rs

```diff
@@ -146,25 +156,43 @@ pub fn sats(amount: u64) -> Amount {
     Amount::from_sats(amount)
 }
 
+pub mod amount {
+    pub mod serde {
+        pub mod as_sat {
+            //! Serialize and deserialize [`Amount`](crate::Amount) as integers
+            //! denominated in satoshi. Use with
+            //! `#[serde(with = "amount::serde::as_sat")]`.
+
+            use serde::{Deserialize, Deserializer, Serialize, Serializer};
+
+            pub fn serialize<S: Serializer>(a: &crate::Amount, s: S) -> Result<S::Ok, S::Error> {
+                u64::serialize(&a.sats(), s)
+            }
+
+            pub fn deserialize<'d, D: Deserializer<'d>>(d: D) -> Result<crate::Amount, D::Error> {
+                Ok(crate::Amount::from_sats(u64::deserialize(d)?))
+            }
+        }
```

Comment:
I don't think we should make it easy to accidentally throw away msat precision like that. E.g. in any JSON API call amounts will be rounded down that way.